cpp/src/io/json/json_column.cu

```diff
@@ -560,6 +592,25 @@ void make_device_json_column(device_span<SymbolT const> input,
     auto& parent_col = it->second.get();
     bool replaced    = false;
     if (mapped_columns.count({parent_col_id, name}) > 0) {
+      // If mixed type is enabled, make both of them as str, merge them.
+      // all its child columns will be ignored from parsing.
+      if (is_mixed_type_as_string_enabled) {
+        // VAL/STR or STRUCT or LIST
+        auto old_col_id = mapped_columns[{parent_col_id, name}];
+
+        is_mixed_string_column[this_col_id] = 1;
+        is_mixed_string_column[old_col_id]  = 1;
+        // if old col type (not cat) is not string/val, replace with string.
+        auto& col = columns.at(old_col_id).get();
+        if (col.type != json_col_t::StringColumn) {
+          // TODO: old_col_id or this_col_id ? affects max_rowoffsets, need more tests.
```

Comment:
Reminder of this TODO. Is this intended to be fixed in this PR?

Reply:
removed TODO and added a unit test.